Documentation/contributing.rst

```diff
@@ -764,6 +764,26 @@ To run this you can use the following command:
     ginkgo  -v -- --cilium.provision=false --cilium.SSHConfig="cat ssh-config"
 
 
+End to End Virtual VMs
+~~~~~~~~~~~~~~~~~~~~~~
+
+The end to end servers are defined in ``test/Vagrantfile``, these servers have a
```

Comment:
"end to end server" does not make sense. This is a run-on sentence that should be split up into multiple sentences.

Maybe:

"The VMs used for testing are defined in ``test/Vagrantfile``. There are a variety of configuration options that can be passed as environment variables:"